server/routes/api/landing_page.py

```diff
@@ -455,23 +476,29 @@ def data(dcid):
                 del spec_and_stat[category][topic]
             else:
                 spec_and_stat[category][topic] = filtered_charts
-        if not spec_and_stat[category]:
-            del spec_and_stat[category]
+        if new_landing_page_cache_flag == "false":
+            if not spec_and_stat[category]:
+                del spec_and_stat[category]
+        else:
+            # Don't delete a category if it is in the valid categories list.
+            if not spec_and_stat[category] and category not in raw_page_data[
+                    "validCategories"][dcid]["category"]:
```

Comment:
Wonder if it's better to check before accessing maps to not throw exceptions and handle gracefully? (e.g., if code runs against an import-group not having a change)

Reply:
Done